### PR TITLE
Leaderboard styling

### LINKED ISSUE #15: Scrollbar appears while hiding leaderboard
https://user-images.githubusercontent.com/248741/129833794-514f2084-c3a3-4be1-90c5-aedfa16cd940.mov

Maybe adding `overflow: hidden;` somewhere will fix this. 👀

### PR DESCRIPTION
**Pull request by Noxturnix**
- [x] Pointer cursor on hover the leaderboard header
- [x] Gaps between top-three `span` elements
- [x] Top-three element should not be hidden on small screens
- [x] Fix #15
- [x] Fix leaderboard overflow on small screens ~(cancel "Top-three element should not be hidden on small screens" task?)~ see https://github.com/Noxturnix/popyut/pull/2

Do not merge until all features/issues are implemented/resolved :D

### CODE REVIEW

**Comment by Noxturnix** [comment]
Seems the top-three element overflows on small screens. This needs to be fixed. Maybe remove the top-three box? or ignore the "Top-three element should not be hidden on small screens" task

Also, I have hard-coded some values to the CSS. This also isn't good. This should be also fixed
```
/* TODO: Make height not be estimated? */
max-height: 5.25rem; /* Estimate height of modalHeader */
```
```
/* TODO: Make height not be estimated? */
max-height: calc(var(--max-leaderboard-height) + 3.05rem) /* 3.05rem is an estimated value */;
```

**Comment by bossoq** [comment]
@Noxturnix Try to fix top three on smaller screen by adding padding krub.
Please review this PR. https://github.com/Noxturnix/popyut/pull/1

**Comment by narze** [comment]
LGTM for me
@scomma please review when you have time (and all checkboxes are checked)

**Comment by scomma** [comment]
Idea: we don't need to show all of top 3 on small screens, only top position is fine? @narze @Noxturnix

**Comment by Noxturnix** [comment]
I'd also like the `modalHeader.top-three` to just have the 1st place too

**Comment by bossoq** [comment]
Add hide top three based on screen size.
There is 3 breakpoint:
 1. < 1000px : hide third place
 2. < 700px : add padding to avoid element overlay with footer
 3. < 680px : hide second place

Plrease review this https://github.com/Noxturnix/popyut/pull/2

**Comment by scomma** [comment]
@bossoq would you mind using the tailwind [responsive classes](https://tailwindcss.com/docs/responsive-design) instead? This is so we use the same breakpoints in the whole app, and to keep with the tailwind spirit of writing minimal custom CSS 😁

**Comment by Noxturnix** [comment]
I think everything looks perfect now. We will also need to clean up some codes, but for me, it's safe to merge now

File: src/routes/index.svelte [hunk 1]
```diff
@@ -393,43 +393,78 @@
...
     bottom: 0;
-    /* left: 50%; */
-    /* transform: translate(-50%, -50%); */
-    /* min-height: 30%; */
     min-width: 50%;
+    width: fit-content;
     background-color: white;
+    transition: max-height 0.4s;
     border-radius: 10px 10px 0 0;
+    overflow-y: hidden;
```
Comment (narze) [inline src/routes/index.svelte:411]: Can use `overflow-x-hidden` Tailwind class

**Comment by bossoq** [comment]
Remove most of custom css and use tailwind class instead krub https://github.com/Noxturnix/popyut/pull/3/

File: src/routes/index.svelte [hunk 2]
```diff
@@ -393,43 +393,78 @@
...
 
-  .modalBody.open {
-    overflow-y: scroll;
-    padding: 0.75rem;
-    max-height: 60vh; /* Change this value according to your needs */
+  #top-three {
+    gap: 0.75rem;
+  }
+
+  @media only screen and (max-width: 1000px) {
```
Comment (narze) [inline src/routes/index.svelte]: For media queries you can use Tailwind breakpoints prefix. Ref: https://tailwindcss.com/docs/responsive-design

and I think we should rather use `min-width` than `max-width` like Tailwind does in the standard breakpoints to encourage using [Mobile-first system](https://tailwindcss.com/docs/responsive-design#mobile-first), so the rules should be changed from:

```
...other -> supported desktop first (1001+)
max-width: 680px -> supported small screen (0-680)
max-width: 1000px -> supported medium screen (681-1000)
```

to:

```
...other -> supported mobile first (0-679)
min-width: 680px -> (680-999) -> use `sm` or `md` breakpoint
min-width: 1000px -> (1000+) -> use `lg` breakpoint
```

And then convert then to standard Tailwind breakpoints with class
Example : 

#top-3 to be hidden on small screens upto `lg`, use `class="hidden lg:block"`

Comment (narze) [reply src/routes/index.svelte]: Or maybe I will try to fix them tonight 😅

**Comment by Noxturnix** [comment]
I'm going to take some rest for now. I will merge and review pull requests tomorrow :grin:

**Comment by narze** [comment]
@Noxturnix I will take it from here, thank you for contribution! 🙌🏻

**Comment by scomma** [comment]
@narze lmk when you finish, I have more tweaks planned for the leaderboard and conflict will be intense 🦹

**Comment by narze** [comment]
@scomma ok, just started doing it right now, should not take more than 15 minutes

**Pull request merged**